react/features/overlay/components/OverlayContainer.js

```diff
@@ -26,262 +50,93 @@ class OverlayContainer extends Component {
-                = filmstripOnly
-                    ? UserMediaPermissionsFilmstripOnlyOverlay
-                    : UserMediaPermissionsOverlay;
-            props = {
-                browser: this.props._browser
-            };
-        } else if (this.props._callOverlayVisible) {
-            overlayComponent = CallOverlay;
-        }
+        return overlay ? React.createElement(overlay, {}) : null;
+    }
+}
 
-        return (
-            overlayComponent
-                ? React.createElement(overlayComponent, props)
-                : null);
+/**
+ * Returns the list of overlay React {@link Component} types to be rendered by
+ * {@code OverlayContainer}. The list is lazily initialized the first time it is
+ * required in order to works around side effects of circular imports.
+ *
+ * @param {boolean} filmstripOnly - The indicator which determines whether the
+ * user interface is filmstrip-only.
+ * @returns {Array} The list of overlay React {@code Component} types to be
+ * rendered by {@code OverlayContainer}.
+ */
+function _getOverlays(filmstripOnly) {
+    let overlays;
+
+    if (filmstripOnly) {
+        if (!(overlays = _filmstripOnlyOverlays)) {
+            overlays = _filmstripOnlyOverlays = [
+                PageReloadFilmstripOnlyOverlay,
+                SuspendedFilmstripOnlyOverlay,
+                UserMediaPermissionsFilmstripOnlyOverlay
+            ];
+        }
+    } else if (!(overlays = _nonFilmstripOnlyOverlays)) {
+        overlays = _nonFilmstripOnlyOverlays = [
+            PageReloadOverlay,
+            SuspendedOverlay,
+            UserMediaPermissionsOverlay,
+            CallOverlay
+        ];
     }
+
+    return overlays;
 }
 
 /**
- * Maps (parts of) the redux state to the associated OverlayContainer's props.
+ * Maps (parts of) the redux state to the associated {@code OverlayContainer}'s
+ * props.
  *
  * @param {Object} state - The redux state.
+ * @private
  * @returns {{
- *     _browser: string,
- *     _callOverlayVisible: boolean,
- *     _connectionEstablished: boolean,
- *     _haveToReload: boolean,
- *     _isNetworkFailure: boolean,
- *     _isMediaPermissionPromptVisible: boolean,
- *     _reason: string,
- *     _suspendDetected: boolean
+ *     overlay: ?Object
  * }}
- * @private
  */
 function _mapStateToProps(state) {
-    const stateFeaturesOverlay = state['features/overlay'];
+    // XXX In the future interfaceConfig is expected to not be a global variable
+    // but a redux state like config. Hence, the variable filmStripOnly
+    // naturally belongs here in preparation for the future.
+    const filmstripOnly
+        = typeof interfaceConfig === 'object' && interfaceConfig.filmStripOnly;
+    let overlay;
+
+    for (const o of _getOverlays(filmstripOnly)) {
+        // react-i18n / react-redux wrap components and thus we cannot access
+        // the wrapped component's static methods directly.
+        const component = o.WrappedComponent || o;
```

Comment:
@saghul, could you please confirm that `o.WrappedComponent || o` is enough for `PageReloadOverlay` which uses both `translate` and `connect`: `translate(connect(abstractMapStateToProps)(PageReloadOverlay))`? I tried to cause the render of `PageReloadOverlay` but I didn't succeed. This is the only thing I need in order to merge this PR.

Reply:
I can confirm this works as expected. I tested it as follows:
- join a conference on my deployment
- restart jicofo
- the overlay shows correctly